modules/network/src/integrationTest/java/org/apache/ignite/internal/network/netty/ItConnectionManagerTest.java

```diff
@@ -296,6 +300,61 @@ public void testStopTwice() throws Exception {
         server.close();
     }
 
+    @Test
+    public void testOneConnectionType() throws Exception {
+        int size = 10000000;
+        char[] chars = new char[size];
+        Arrays.fill(chars, 'a');
+        String bigText = new String(chars);
+
+
+        ConnectionManagerWrapper server1 = startManager(4000);
+        ConnectionManagerWrapper server2 = startManager(4001);
+
+        NettySender sender = server1.openChannelTo(server2, ChannelType.DEFAULT).get(3, TimeUnit.SECONDS);
+
+
+        TestMessage bigMessage = messageFactory.testMessage().msg(bigText).build();
+        TestMessage msg = messageFactory.testMessage().msg("test").build();
+
+        CompletableFuture<Void> send1 = sender.send(new OutNetworkObject(bigMessage, Collections.emptyList()));
+        CompletableFuture<Void> send2 = sender.send(new OutNetworkObject(msg, Collections.emptyList()));
+
+        assertThat(send2, willCompleteSuccessfully());
+        assertThat(send1.isDone(), is(true));
+
+        server1.close();
+        server2.close();
+    }
+
+    @Test
+    public void testTwoConnectionTypes() throws Exception {
+        int size = 1000000000;
+        char[] chars = new char[size];
+        Arrays.fill(chars, 'a');
+        String bigText = new String(chars);
+
+
+        ConnectionManagerWrapper server1 = startManager(4000);
+        ConnectionManagerWrapper server2 = startManager(4001);
+
+        NettySender sender1 = server1.openChannelTo(server2, ChannelType.DEFAULT).get(3, TimeUnit.SECONDS);
+        NettySender sender2 = server1.openChannelTo(server2, ChannelType.TEST).get(3, TimeUnit.SECONDS);
+
+
+        TestMessage bigMessage = messageFactory.testMessage().msg(bigText).build();
+        TestMessage msg = messageFactory.testMessage().msg("test").build();
+
+        CompletableFuture<Void> send1 = sender1.send(new OutNetworkObject(bigMessage, Collections.emptyList()));
+        CompletableFuture<Void> send2 = sender2.send(new OutNetworkObject(msg, Collections.emptyList()));
+
+        assertThat(send2, willCompleteSuccessfully());
+        assertThat(send1.isDone(), is(false));
+
+        server1.close();
+        server2.close();
```

Comment:
Could you close the servers in AfterEach block? In case the assert is wrong they will hang.